File: back-end-api/src/main/java/com/yueying/backendapi/model/domain/request/UpdateUserInfoRequest.java
package com.yueying.backendapi.model.domain.request;

import lombok.Data;

import java.io.Serial;
import java.io.Serializable;

/**
 * 用户信息修改请求体
 * @author <a href="mengnalidou.icu">mengnali_dou</a>
 */
@Data
public class UpdateUserInfoRequest implements Serializable {

    @Serial
    private static final long serialVersionUID = -3717320337916839138L;

    /**
     * 用户ID
     */
    private long userId;

    /**
     * 用户名
     */
    private String userName;

    /**
     * 账号
     */
    private String userAccount;

    /**
     * 头像URL
     */
    private String avatarUrl;

    /**
     * 性别
     */
    private int gender;

    /**
     * 电话
     */
    private String phone;

    /**
     * email
     */
    private String email;
}

File: front-end/src/generate/request/update-user-info-request.ts
/**
 * 更新用户信息请求体
 * @author <a href="mengnalidou.icu">mengnali_dou</a>
 */
export interface UpdateUserInfoRequest {
	// id
	userId: number;

	// 用户名
	userName: string;

	// 账号
	userAccount: string;

	// 头像
	avatarUrl: string;

	// 性别
	gender: number;

	// 电话
	phone: string;

	// email
	email: string;
}
